helm/external-dns-app/templates/_helpers.tpl

```diff
@@ -207,8 +207,13 @@ Set Giant Swarm podAnnotations.
 Set Giant Swarm serviceAccountAnnotations.
 */}}
 {{- define "giantswarm.serviceAccountAnnotations" -}}
-{{- if and (or (eq .Values.provider "aws") (eq .Values.provider "capa")) (eq .Values.aws.irsa "true") (eq .Values.aws.access "internal") (not (hasKey .Values.serviceAccount.annotations "eks.amazonaws.com/role-arn")) }}
+{{- if and (eq .Values.aws.irsa "true") (eq .Values.aws.access "internal") (not (hasKey .Values.serviceAccount.annotations "eks.amazonaws.com/role-arn")) }}
+{{- if (eq .Values.provider "aws") }}
 {{- $_ := set .Values.serviceAccount.annotations "eks.amazonaws.com/role-arn" (tpl "arn:aws:iam::{{ .Values.aws.accountID }}:role/{{ template \"aws.iam.role\" . }}" .) }}
+{{- end}}
+{{- if and (or (eq .Values.provider "capa")) }}
```

Comment:
We still need to verify aws access internal. The reason is customer can also use "external" which uses ENV to set AWS credentials 

```suggestion
{{- else if and (eq .Values.provider "capa") (eq .Values.aws.access "internal") }}
```